### PR TITLE
final_state_tuple is a Tuple

### LINKED ISSUE #2503: generator can't return size (using alternating LSTM)
**Describe the bug**
I'm trying to use an alternating LSTM with the following config:

```
"loglines_encoder": {
            "type": "alternating_lstm",
            "recurrent_dropout_probability": 0.44,
            "input_size": 100,
            "hidden_size": 128,
            "num_layers": 2,
            "use_input_projection_bias": false
}
```
My `loglines_encoder` is of type `Seq2VecEncoder`. I get the following error:

```
Traceback (most recent call last):
  File "/apps/python3/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/apps/python3/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/run.py", line 18, in <module>
    main(prog="allennlp")
  File "/apps/python3/lib/python3.6/site-packages/allennlp/commands/__init__.py", line 72, in main
    args.func(args)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/commands/train.py", line 111, in train_model_from_args
    args.force)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/commands/train.py", line 144, in train_model_from_file
    return train_model(params, serialization_dir, file_friendly_logging, recover, force)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/commands/train.py", line 361, in train_model
    metrics = trainer.train()
  File "/apps/python3/lib/python3.6/site-packages/allennlp/training/trainer.py", line 787, in train
    train_metrics = self._train_epoch(epoch)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/training/trainer.py", line 515, in _train_epoch
    loss = self.batch_loss(batch, for_training=True)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/training/trainer.py", line 434, in batch_loss
    output_dict = self.model(**batch)
  File "/apps/python3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/data/source_to_tags/code/mux/models/model.py", line 167, in forward
    encoded_logline = self.loglines_encoder(embedded_logline, logline_mask)
  File "/apps/python3/lib/python3.6/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/apps/python3/lib/python3.6/site-packages/allennlp/modules/seq2vec_encoders/pytorch_seq2vec_wrapper.py", line 79, in forward
    num_layers_times_directions, num_valid, encoding_dim = state.size()
AttributeError: 'generator' object has no attribute 'size'
```

Seems like `seq2vec_wrapper.py` needs to be updated?

**To Reproduce**
Use a `Seq2VecEncoder` with the config above. 

**System (please complete the following information):**
 - OS: Linux
 - Python version: 3.6.5
 - AllenNLP version: v0.8.1
 - PyTorch version: 1.0.0

### PR DESCRIPTION
**Pull request by vidurj**
Fixes #2503. The `final_state_tuple` in the `StackedAlternatingLstm` was a generator instead of a tuple.

### CODE REVIEW

File: allennlp/modules/stacked_alternating_lstm.py [hunk 1]
```diff
@@ -106,5 +106,5 @@ def forward(self,  # pylint: disable=arguments-differ
             output_sequence, final_state = layer(output_sequence, state)
             final_states.append(final_state)
 
-        final_state_tuple = (torch.cat(state_list, 0) for state_list in zip(*final_states))
+        final_state_tuple = tuple(torch.cat(state_list, 0) for state_list in zip(*final_states))
```
Comment (brendan-ai2) [inline allennlp/modules/stacked_alternating_lstm.py]: Could you please add a return type on line 74 and correct the `Returns` section of the docstring?

Comment (vidurj) [reply allennlp/modules/stacked_alternating_lstm.py]: Done, thanks.

File: allennlp/tests/modules/seq2vec_encoders/pytorch_seq2vec_wrapper_test.py [hunk 2]
```diff
@@ -113,3 +114,14 @@ def test_wrapper_raises_if_batch_first_is_false(self):
         with pytest.raises(ConfigurationError):
             lstm = LSTM(bidirectional=True, num_layers=3, input_size=3, hidden_size=7)
             _ = PytorchSeq2VecWrapper(lstm)
+
+    def test_wrapper_works_with_alternating_lstm(self):
+        model = PytorchSeq2VecWrapper(
+                StackedAlternatingLstm(input_size=100,
```
Comment (DeNeutoy) [inline allennlp/tests/modules/seq2vec_encoders/pytorch_seq2vec_wrapper_test.py]: No need for the test fixture to be this size, it makes them slow - just use 4,5 etc.

Comment (vidurj) [reply allennlp/tests/modules/seq2vec_encoders/pytorch_seq2vec_wrapper_test.py]: Good point! Thanks.

**Pull request merged**